snappy_device_agents/__init__.py

```diff
@@ -227,11 +172,10 @@ def get_local_ip_addr():
     :return ipaddr:
         Returns the ip address of this system
     """
-    gateways = netifaces.gateways()
-    default_interface = gateways["default"][netifaces.AF_INET][1]
-    ipaddr = netifaces.ifaddresses(default_interface)[netifaces.AF_INET][0][
-        "addr"
-    ]
+    # Use SOCK_DGRAM since we don't need to send any data and to avoid timeout
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    sock.connect(("10.0.0.0", 0))
+    ipaddr = sock.getsockname()[0]
```

Comment:
To be entirely clean shouldn't the socket be closed?

Reply:
good idea, or even better yet, sockets can use a context manager :) How about now?

Reply:
Yes much nicer!